src/core/thread/mle_router.cpp

```diff
@@ -2522,6 +2543,31 @@ void MleRouter::HandleChildUpdateRequest(const Message &         aMessage,
         ExitNow(error = OT_ERROR_PARSE);
     }
 
+#if OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
+    // CSL
+    if (child->IsCslSynchronized())
+    {
+        CslChannelTlv cslChannel;
+        uint32_t      cslTimeout;
+
+        if (Tlv::FindUint32Tlv(aMessage, Tlv::kCslTimeout, cslTimeout) == OT_ERROR_NONE)
+        {
+            child->SetCslSyncTimeout(cslTimeout);
+        }
+
+        if (Tlv::FindTlv(aMessage, Tlv::kCslChannel, sizeof(cslChannel), cslChannel) == OT_ERROR_NONE)
+        {
+            child->SetCslChannel(static_cast<uint8_t>(cslChannel.GetChannel()));
+        }
+        else
+        {
+            child->SetCslChannel(Get<Mac::Mac>().GetPanChannel());
+        }
+
+        tlvs[tlvslength++] = Tlv::kSourceAddress;
```

Comment:
Reading the code above this, look like we always add  `Tlv::kSourceAddress` few lines above this from
```
    // Find Child
    aMessageInfo.GetPeerAddr().GetIid().ConvertToExtAddress(extAddr);
    child = mChildTable.FindChild(extAddr, Child::kInStateAnyExceptInvalid);

    tlvs[tlvslength++] = Tlv::kSourceAddress;
```

Is there a reason we are re-adding it here with `tlvs[tlvslength++] = Tlv::kSourceAddress;`

This may actually cause the same TLV to be added twice in the message from `SendChildUpdateResponse()`.